### PR TITLE
Add build.target option to docker_image module

### PR DESCRIPTION
**Pull request by Pluggi**
##### SUMMARY

Add support for --target option to docker_image module when building an image.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

`docker_image`

### CODE REVIEW

File: lib/ansible/modules/cloud/docker/docker_image.py [hunk 1]
```diff
@@ -126,6 +126,11 @@
             be set in the container being built.
           - Needs Docker SDK for Python >= 3.7.0.
         type: bool
+      target:
+        description:
+          - When building an image specifies an intermediate build stage by
+            name as a final stage for the resulting image.
+        type: str
```
Comment (felixfontein) [inline lib/ansible/modules/cloud/docker/docker_image.py:133]: You need to add `version_added: "2.9"`. Also, you need a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to).

**Review by felixfontein — APPROVED** [review]
Looks good! Two minor things are missing. Also, would you mind adding a simple integration test for this? See [here](https://github.com/ansible/ansible/tree/devel/test/integration/targets/docker_image/tasks/tests/) for where the current tests are.

**Comment by Pluggi** [comment]
Thank you!
I wrote the integration tests but am struggling with ansible-test to get them to run.
Do you want me to push it?

File: test/integration/targets/docker_image/tasks/tests/options.yml [hunk 2]
```diff
@@ -290,3 +290,28 @@
...
+- name: cleanup
+  docker_image:
+    name: "{{ iname }}"
+    state: absent
+    force_absent: yes
+
+- assert:
+    that:
+    - dockerfile_2 is changed
+    - dockerfile_2['image']['Config']['WorkingDir'] == '/first'
```
Comment (felixfontein) [inline test/integration/targets/docker_image/tasks/tests/options.yml]: You can simplify this to
```suggestion
    - dockerfile_2.image.Config.WorkingDir == '/first'
```

File: changelogs/fragments/58487-docker_image_build.target_option.yaml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+- "docker_image - Add build.target option."
```
Comment (felixfontein) [inline changelogs/fragments/58487-docker_image_build.target_option.yaml]:
```suggestion
- "docker_image - Add ``build.target`` option."
```

**Comment by felixfontein** [comment]
Pushing it is usually easier than getting the docker_* tests run locally.

**Comment by Pluggi** [comment]
All the issues are fixed and I squashed all the commits.

**Comment by felixfontein** [comment]
There's no need to squash the commits, it's easier to follow what happened in a PR if they aren't :) Also, when this PR is merged, they will be squashed anyway.

**Review by felixfontein — APPROVED** [review]
shipit

**Comment by WojciechowskiPiotr** [comment]
shipit

**Review by resmo — APPROVED** [review]
shipit

**Comment by felixfontein** [comment]
This is really strange, we already have two maintainer shipit's and still ansibot isn't adding the `shipit` label. Also, my supershipit powers should ensure that one `shipit` by me is already enough.

bot_status

**Comment by felixfontein** [comment]
Interesting, ansibot seems to ignore both @resmo and me.

**Comment by felixfontein** [comment]
@Pluggi I want to keep this PR open a few more days, so the ansibot people can try to debug this ansibot problem. I hope you don't mind. It will get merged eventually, and as a new feature, it won't be available before Ansible 2.9 anyway, so it doesn't really matter whether it's merged now or in a week :)

In any way, thanks for implementing this!

**Comment by mkrizek** [comment]
bot_status

**Comment by mkrizek** [comment]
bot_status

**Comment by felixfontein** [comment]
@Pluggi thanks for implementing this! The feature will show up in Ansible 2.9, to be released by the end of this year.
@WojciechowskiPiotr @resmo thanks for reviewing!
and @mkrizek thanks for fixing ansibot :)

**Pull request merged**